tests/test_execution.py

```diff
@@ -1,15 +1,14 @@
 import pytest
-import copy
 
 import substra
 
 import substratest as sbt
 
 
-def test_tuples_execution_on_same_node(data_network):
+def test_tuples_execution_on_same_node(global_execution_env):
     """Execution of a traintuple, a following testtuple and a following traintuple."""
-    factory, network = data_network
-    session = copy.deepcopy(network.sessions[0])
+    factory, network = global_execution_env
+    session = network.sessions[0].copy()
```

Comment:
I haven't thought about this earlier, sorry. The copy could be done directly inside the fixture: it will prevent fixture user to modify its content and thus simplify the fixture usage.

Reply:
This is a session fixture that is only executed once. Returning a copy in the fixture means that all tests will share the same copy. If we want to isolate the sessions during the tests, we'll have to make a copy in each test.

Reply:
Good point!